src/app/services/sources.ts

```diff
@@ -1,34 +1,57 @@
 import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
 import type KumaApi from '@/services/kuma-api/KumaApi'
-import type {
-  PaginatedApiListResponse as CollectionResponse,
-} from '@/types/api.d'
-import type {
-  ServiceInsight,
-} from '@/types/index.d'
-
-type MeshParams = {
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type { ExternalService, ServiceInsight } from '@/types/index.d'
+
+type CollectionParams = {
   mesh: string
 }
+
+type DetailParams = CollectionParams & {
+  name: string
+}
+
 type PaginationParams = {
   size: number
   page: number
 }
+
 type Closeable = { close: () => void }
+
+export type ServiceInsightSource = DataSourceResponse<ServiceInsight>
 export type ServiceInsightCollection = CollectionResponse<ServiceInsight>
 export type ServiceInsightCollectionSource = DataSourceResponse<ServiceInsightCollection>
 
+export type ExternalServiceSource = DataSourceResponse<ExternalService>
+
 export const sources = (api: KumaApi) => {
   return {
-    '/:mesh/services': async (params: MeshParams & PaginationParams, source: Closeable) => {
+    '/:mesh/service-insights': async (params: CollectionParams & PaginationParams, source: Closeable) => {
```

Comment:
Why change this name?

Reply:
Because the services term is pretty overloaded in the UI application due to there being services in the context of our DI system, services in the context of the application domain (`ServiceInsight`, `ExternalService`). Personally, I found it easiest to formulate these paths roughly after the names of the types/resources we access through them.